t/t1200-stats-basic.t

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+#
+
+test_description='Test stats collection and sending'
+
+. `dirname $0`/sharness.sh
+
+test_under_flux 2
+
+test_expect_success 'content-cache packets recieved' '
+	FLUX_FRIPP_STATSD=0.0.0.0:9999 flux start socat UDP-RECVFROM:9999 -
```

Comment:
Oops, `socat` is not installed in the ci images (nor on my system), so this test is failing.

Also, how will the `socat` know to exit so the test stops?

Here's one idea. You can create a UDP listener in Python for use with the tests. Here's a basic demo that waits for just one packet and exits. (You'd want to add a timeout to this test to detect no packets and fail.

This version doesn't check the format of the packet, but this could easily be added. The script or different scripts could be enhanced to check for properly formatted or expected stats strings, so it might be a good base to start from:

```diff
diff --git a/t/t1200-stats-basic.t b/t/t1200-stats-basic.t
index 1e9f2aa3f..2bfef452b 100755
--- a/t/t1200-stats-basic.t
+++ b/t/t1200-stats-basic.t
@@ -7,8 +7,18 @@ test_description='Test stats collection and sending'
 
 test_under_flux 2
 
+test_expect_success 'create udp client program' '
+       cat <<-EOF >listen.py &&
+       import socket
+       s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+       s.bind(("0.0.0.0", 9999))
+       print(s.recvfrom(1024))
+       EOF
+       chmod +x listen.py
+'
+
 test_expect_success 'content-cache packets recieved' '
-       FLUX_FRIPP_STATSD=0.0.0.0:9999 flux start socat UDP-RECVFROM:9999 -
+       FLUX_FRIPP_STATSD=0.0.0.0:9999 flux start flux python listen.py
 '
 
 test_done

```

Reply:
```
 grondo@asp:~/git/f.git/t$ ./t1200-stats-basic.t  -d -v
sharness: loading extensions from /home/grondo/git/f.git/t/sharness.d/01-setup.sh
sharness: loading extensions from /home/grondo/git/f.git/t/sharness.d/flux-sharness.sh
sharness: loading extensions from /home/grondo/git/f.git/t/sharness.d/01-setup.sh
sharness: loading extensions from /home/grondo/git/f.git/t/sharness.d/flux-sharness.sh
expecting success: 
	cat <<-EOF >listen.py &&
	import socket
	s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
	s.bind(("0.0.0.0", 9999))
	print(s.recvfrom(1024))
	EOF
	chmod +x listen.py

ok 1 - create udp client program

expecting success: 
	FLUX_FRIPP_STATSD=0.0.0.0:9999 flux start flux python listen.py

(b'flux.0.content-cache.size:1360|g\nflux.0.content-cache.dirty:0|g\nflux.0.content-cache.count:10|g\nflux.0.content-cache.flush-batch-count:0|g\nflux.0.content-cache.valid:10|g', ('127.0.0.1', 46371))
ok 2 - content-cache packets recieved

# passed all 2 test(s)
1..2

```

Reply:
`socat` was exiting after receiving the first packet when I was testing. Not sure if it's safe to assume that'll always happen.

Reply:
Ah, apologies I didn't know that (and didn't have socat to test with :wink:)

If we can get the python script to parse and validate the packets, then that will really solidify the testing.
You could add options to wait for N packets, or wait for a value to appear (e.g. if you wanted to ensure 1 running job was counted, etc.)